test/extensions/filters/listener/tls_inspector/tls_inspector_test.cc

```diff
@@ -282,6 +311,39 @@ TEST_P(TlsInspectorTest, InlineReadSucceed) {
   EXPECT_EQ(Network::FilterStatus::Continue, filter_->onAccept(cb_));
 }
 
+// Test that the filter is activated when the dst port matches a port on the whitelist
+// dst port = 3000, whitelist = {3000}
+TEST_P(TlsInspectorTest, WhitelistedDstPortIsInspected) {
+  init(makeConfigWithWhitelistedPort(3000));
+  std::vector<uint8_t> data;
+
+  // Use 100 bytes of zeroes. This is not valid as a ClientHello.
+  data.resize(100);
+
+  EXPECT_CALL(os_sys_calls_, recv(42, _, _, MSG_PEEK))
+      .WillOnce(
+          Invoke([&data](os_fd_t, void* buffer, size_t length, int) -> Api::SysCallSizeResult {
+            ASSERT(length >= data.size());
+            memcpy(buffer, data.data(), data.size());
+            return Api::SysCallSizeResult{ssize_t(data.size()), 0};
+          }));
+  EXPECT_CALL(cb_, continueFilterChain(true));
+  file_event_callback_(Event::FileReadyType::Read);
```

Comment:
is this needed?

Reply:
Yes, the tests fail without it. The callback invocation is what triggers the call to `continueFilterChain`.

BTW, this test case is mostly a copy of `NotSsl`.